plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/Simplify.hs

```diff
@@ -0,0 +1,21 @@
+{-# LANGUAGE LambdaCase #-}
+-- | Very basic simplifications on UPLC.
+module UntypedPlutusCore.Transform.Simplify
+    ( simplifyTerm
+    , simplifyProgram
+    ) where
+
+import           UntypedPlutusCore.Core
+
+import           Control.Lens
+
+simplifyProgram :: Program name uni fun a -> Program name uni fun a
+simplifyProgram (Program a v t) = Program a v $ simplifyTerm t
+
+simplifyTerm :: Term name uni fun a -> Term name uni fun a
+simplifyTerm = transformOf termSubterms processTerm
```

Comment:
Hold on.  Does this get `Force (Force (Delay (Delay ...))))?  It's not immediately obvious to me if that's the case, despite having looked up `transformOf`.

Reply:
It does: there's a test to that effect! `transformOf` processes subterms first, which means that you zap the inner terms before looking at the outer ones.

Reply:
Yeah, that occurred to me just after I went out for a walk.  I'll walk before typing in future.